### ISSUE TITLE
RFP - tldr

### ISSUE THREAD

**Issue by StewAlexander-com** [chocolatey-community/chocolatey-package-requests#707]
Checklist
- [x] The package I am requesting does not already exist on https://chocolatey.org/packages;
- [x] There is no open issue for this package;
- [x] The issue title starts 'RFP - ';
- [x] The download URL is public and not locked behind a paywall / login;
Package Details
Software project URL : See https://github.com/tldr-pages/tldr
Direct download URL for the software / installer : See https://github.com/tldr-pages/tldr
Software summary / short description:

TL;DR is an index of command descriptions and examples that gets to the point, details see here:
https://github.com/tldr-pages/tldr

See [below comment](https://github.com/chocolatey-community/chocolatey-package-requests/issues/707#issuecomment-546995354) - the Python version would the best one to package up and make available for Chocolatey users.

**Comment by ionred** [comment]
Researching

**Comment by AdmiringWorm** [comment]
@StewAlexanderACC 
can you please update the issue with the RFM template, and provide the all the details mentioned in that? Thanks

**Comment by StewAlexanderACC** [comment]
@AdmiringWorm 

My pleasure, where is this form you speak of?

**Comment by StewAlexanderACC** [comment]
-->
## Checklist

- [ x] The package I am requesting does not already exist on https://chocolatey.org/packages;
- [x ] There is no open issue for this package;
- [x ] The issue title starts 'RFP - ';
- [N/A ] The download URL is public and not locked behind a paywall / login;

## Package Details

Software project URL : **none, needs to be made**
Direct download URL for the software / installer : N/A
Software summary / short description:

TL;DR is a index of command descriptions that gets to the point, details see here: 
https://github.com/tldr-pages/tldr



Using _**Gerhkin Acceptance Criteria**_:

Scenario — Individual wants a concise description and example of a command
Given —  A need to lookup concise information and or find a specific command for a process
When —  They type tldr <command name> at a prompt
Then — Collects the specific tldr command information here: https://github.com/tldr-pages/tldr/tree/master/pages/windows
And —  Presents this information in a concise way to either the command line, and or application screen

**Comment by pauby** [comment]
Is this:

1. A request to create tldr pages for Chocolatey on the tldr GH repo? Or 
2. A request to create a package that ends up with a working tldr command on Windows?

**Comment by StewAlexanderACC** [comment]
@pauby #2 specifically - chocolatey commands are already in the repo (see here:https://github.com/tldr-pages/tldr/tree/master/pages/windows)

**Comment by pauby** [comment]
@StewAlexanderACC Thanks for clarifying that. As soon as you have [updated the template in the original message](https://github.com/chocolatey-community/chocolatey-package-requests/issues/707#issuecomment-546367111) we can make this available.

**Comment by StewAlexanderACC** [comment]
@pauby Updated I believe

**Comment by AdmiringWorm** [comment]
@StewAlexanderACC just so everything will be clear (at least to me), what you are asking about is someone to package one of the clients located in the README? (https://github.com/tldr-pages/tldr#clients)

If so, I would suggest that you mention which one of them you would prefer (if any).

**Comment by StewAlexanderACC** [comment]
@AdmiringWorm 

The bash client won't work natively in the WIndows command line, but if it was **_ported_** then that is what I am talking about

So for instance if someone took the time to port the https://github.com/raylee/tldr client to windows, well that would answer the requirements

Or, build it from scratch using the above as a template, either way it answers my request, thanks

**Comment by StewAlexanderACC** [comment]
@AdmiringWorm 

Actually since this already works with python, it may be easier to just package this: https://github.com/lord63/tldr.py

With python being installed as a dependency

**Comment by StewAlexanderACC** [comment]
@AdmiringWorm all those packages you mention are essentially equal on the front-end (back-end is probably quite different), so whichever is easiest to package, personally I would bet the python package is easiest

**Comment by pauby** [comment]
@StewAlexanderACC Just so we're on the same page. This request is to create a Chocolatey package for tldr. It's not a request to create a new way for tldr to run on Windows. That might already be clear in your head, I just wanted to make sure we are all on the same page.

There have been a few suggestions and most of them sound like they are requesting a new way for tldr to run (as @AdmiringWorm said [above](https://github.com/chocolatey-community/chocolatey-package-requests/issues/707#issuecomment-546407155)):

> The bash client won't work natively in the WIndows command line, but if it was ported then that is what I am talking about
> So for instance if someone took the time to port the https://github.com/raylee/tldr client to windows, well that would answer the requirements
> Or, build it from scratch using the above as a template, either way it answers my request, thanks

That's not what this package requests page is for.

> Actually since this already works with python, it may be easier to modify this https://github.com/lord63/tldr.py for the windows repo and just package it.
> With python being installed as a dependency in the package process

That sounds like it _might_ be depending on what you mean by _modify this for the windows repo_?

**Comment by StewAlexanderACC** [comment]
Well natively  the app probably will only answer back answers to Linux commands, which isn't very useful for some Windows users.

IDK, I was hoping someone could look into this...

As it stands the python tool would be helpful to a subset of Windows users who use Linux

**Comment by pauby** [comment]
@StewAlexanderACC As I said [above](https://github.com/chocolatey-community/chocolatey-package-requests/issues/707#issuecomment-546567881) it sounds very much like you're asking for new app / program / software. This is the wrong place for that.

If the Python tool will work for users of Windows then it would seem to be the best course of action to use that. Do you agree?

Note that I don't use tldr so I have no idea how it works so relying on your answers to guide this.

**Comment by StewAlexanderACC** [comment]
@pauby agreed

For the development concerns, ehh I suppose that will have to be addressed elsewhere.

However the python tool should work "out-of-the-box" for Windows and should be able to be packaged in Chocolatey

**Comment by JanPokorny** [comment]
The situation around tl;dr clients is complicated (and the official README does not help), but judging by what Scoop uses (https://github.com/ScoopInstaller/Main/blob/master/bucket/tldr.json), this is the preferred client: https://isacikgoz.me/tldr/

**Comment by dgalbraith** [comment]
I will pick this up for packaging.

**Comment by pauby** [comment]
@dgalbraith Any update on this?

**Comment by dgalbraith** [comment]
Packaging is complete - just waiting for verification testing to be completed before marking available for review.

**Comment by AdmiringWorm** [comment]
@dgalbraith when you submit a package, please include the link to the package here.
That way we won't have to waste time trying to search for the package (and hopefully find it).

**Comment by dgalbraith** [comment]
Packages are complete and ready for review.

I have added 2 packages here - [tldr](https://www.chocolatey.org/packages/tldr) and [tldr-plusplus](https://www.chocolatey.org/packages/tldr-plusplus).

`tldr` is a meta-package meant to provide an abstraction to the implementation of the client while the [tldr++](https://isacikgoz.me/tldr/) client package is captured as the actual implementation to be installed through a package dependency.

Should a more desirable client be identified in the future the base meta-package could be linked to this through a dependency update.

**Comment by dgalbraith** [comment]
The packages [tldr](https://www.chocolatey.org/packages/tldr) and [tldr-plusplus](https://www.chocolatey.org/packages/tldr-plusplus) have now been reviewed and are available in the community repository.
